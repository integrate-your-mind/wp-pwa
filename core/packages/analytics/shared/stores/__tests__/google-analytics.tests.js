import { unprotect } from 'mobx-state-tree';
import { Stores, itemPost60, itemCat7 } from '../mocks';

let stores;
beforeEach(() => {
  stores = Stores.create({});
  unprotect(stores);
});

describe('Analytics > GoogleAnalytics', () => {
  test('sendPageView', () => {
    window.ga = jest.fn();

    stores.connection.selectedItem = itemPost60;
    stores.analytics.googleAnalytics.sendPageView();
    expect(window.ga).toHaveBeenCalledTimes(2);

    stores.connection.selectedItem = itemCat7;
    stores.analytics.googleAnalytics.sendPageView();
    expect(window.ga).toHaveBeenCalledTimes(4);

    expect(window.ga.mock.calls).toMatchSnapshot();
  });

  test('sendEvent', () => {
    const event = {
      category: 'test/category',
      action: 'test/action',
      label: 'test/label',
    };

    window.ga = jest.fn();
    stores.analytics.googleAnalytics.sendEvent(event);
    expect(window.ga).toHaveBeenCalledTimes(2);
    expect(window.ga.mock.calls).toMatchSnapshot();
  });

  test('ids', () => {
    stores.build = { channel: 'pwa' };
    expect(stores.analytics.googleAnalytics.ids).toMatchSnapshot();
    stores.build = { channel: 'amp' };
    expect(stores.analytics.googleAnalytics.ids).toMatchSnapshot();
  });

  test('trackingOptions', () => {
    const { trackingOptions } = stores.analytics.googleAnalytics;
    let ids;

    stores.build = { channel: 'pwa' };
    ({ ids } = stores.analytics.googleAnalytics);
    expect(ids.map(id => trackingOptions(id))).toMatchSnapshot();

    stores.build = { channel: 'amp' };
    ({ ids } = stores.analytics.googleAnalytics);
    expect(ids.map(id => trackingOptions(id))).toMatchSnapshot();
  });

  test('pageView', () => {
    stores.build = { channel: 'pwa' };
    expect(stores.analytics.googleAnalytics.pageView).toMatchSnapshot();

    stores.build = { channel: 'amp', isAmp: true };
    expect(stores.analytics.googleAnalytics.pageViewAmp).toMatchSnapshot();
  });
});
