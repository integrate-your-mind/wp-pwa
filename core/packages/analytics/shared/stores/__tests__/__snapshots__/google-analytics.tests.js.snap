// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics > GoogleAnalytics ids 1`] = `
Array [
  "UA-12345678-1",
  "UA-12345678-2",
]
`;

exports[`Analytics > GoogleAnalytics ids 2`] = `
Array [
  "UA-12345678-1",
  "UA-12345678-2",
]
`;

exports[`Analytics > GoogleAnalytics pageView 1`] = `
Object {
  "dimension1": "dim1_post60",
  "dimension2": "dim2_post60",
  "location": "https://demo.frontity.test/the-beauties-of-gullfoss/",
  "title": "The Beauties of Gullfoss",
}
`;

exports[`Analytics > GoogleAnalytics pageView 2`] = `
Object {
  "cd1": "dim1_post60",
  "cd2": "dim2_post60",
  "location": "https://demo.frontity.test/the-beauties-of-gullfoss/",
  "title": "The Beauties of Gullfoss",
}
`;

exports[`Analytics > GoogleAnalytics sendEvent 1`] = `
Array [
  Array [
    "tracker_UA_12345678_1.send",
    Object {
      "eventAction": "PWA - test/action",
      "eventCategory": "PWA - test/category",
      "eventLabel": "test/label type: post context: single",
      "hitType": "event",
    },
  ],
  Array [
    "tracker_UA_12345678_2.send",
    Object {
      "eventAction": "PWA - test/action",
      "eventCategory": "PWA - test/category",
      "eventLabel": "test/label type: post context: single",
      "hitType": "event",
    },
  ],
]
`;

exports[`Analytics > GoogleAnalytics sendPageView 1`] = `
Array [
  Array [
    "tracker_UA_12345678_1.send",
    Object {
      "dimension1": "dim1_post60",
      "dimension2": "dim2_post60",
      "hitType": "pageview",
      "location": "https://demo.frontity.test/the-beauties-of-gullfoss/",
      "title": "The Beauties of Gullfoss",
    },
  ],
  Array [
    "tracker_UA_12345678_2.send",
    Object {
      "dimension1": "dim1_post60",
      "dimension2": "dim2_post60",
      "hitType": "pageview",
      "location": "https://demo.frontity.test/the-beauties-of-gullfoss/",
      "title": "The Beauties of Gullfoss",
    },
  ],
  Array [
    "tracker_UA_12345678_1.send",
    Object {
      "dimension1": "dim1_cat7",
      "dimension2": "dim2_cat7",
      "hitType": "pageview",
      "location": "https://demo.frontity.test/wp-cat/photography/page/2",
      "title": "Photography",
    },
  ],
  Array [
    "tracker_UA_12345678_2.send",
    Object {
      "dimension1": "dim1_cat7",
      "dimension2": "dim2_cat7",
      "hitType": "pageview",
      "location": "https://demo.frontity.test/wp-cat/photography/page/2",
      "title": "Photography",
    },
  ],
]
`;

exports[`Analytics > GoogleAnalytics sendPageView with title and location 1`] = `
Array [
  Array [
    "tracker_UA_12345678_1.send",
    Object {
      "dimension1": "dim1_post60",
      "dimension2": "dim2_post60",
      "hitType": "pageview",
      "location": "https://demo.frontity.test/the-beauties-of-gullfoss/2",
      "title": "The Beauties of Gullfoss - page 2",
    },
  ],
  Array [
    "tracker_UA_12345678_2.send",
    Object {
      "dimension1": "dim1_post60",
      "dimension2": "dim2_post60",
      "hitType": "pageview",
      "location": "https://demo.frontity.test/the-beauties-of-gullfoss/2",
      "title": "The Beauties of Gullfoss - page 2",
    },
  ],
]
`;

exports[`Analytics > GoogleAnalytics trackingOptions 1`] = `
Array [
  Object {
    "sendEvents": true,
    "sendPageViews": true,
  },
  Object {
    "sendEvents": true,
    "sendPageViews": true,
  },
]
`;

exports[`Analytics > GoogleAnalytics trackingOptions 2`] = `
Array [
  Object {
    "sendEvents": true,
    "sendPageViews": true,
  },
  Object {
    "sendEvents": true,
    "sendPageViews": false,
  },
]
`;
